Library update-hashes command (#3267)
chainguard-dev/edu: content/chainguard/libraries/javascript/build-configuration.md

<a id="npm"></a>

## Updating lockfile hashes for existing projects

> > **Note**: `chainctl libraries update-hashes` does not currently support authentication through a repository manager. You will need to configure direct access credentials before running the command.

Reviewer: Where do we tell users how to do this? Perhaps we should link to that.

Reviewer: This may apply to a couple of other mentions of configuring direct access credentials in this PR.

If you are migrating an existing JavaScript project to Chainguard Libraries,

> your lockfile contains integrity hashes generated against the npm registry.

Reviewer 2: ```suggestion
your lockfile likely contains integrity hashes generated against the npm registry.
```

> Because Chainguard rebuilds packages from verified source, these hashes differ

Reviewer 3: I'm wondering if this is the best phrasing - it suggests that the reason our packages are different, is due to what sources we use. We generally use the same sources as upstream.

> from Chainguard's checksums and must be updated before reinstalling.

Reviewer 2: ```suggestion
Because Chainguard rebuilds packages from verified source, the hashes from packages previously installed from the upstream npm registry will differ
from Chainguard's, and must be updated before reinstalling.
```

Use the following command to update hashes in place across all supported lockfile formats (`package-lock.json`, `yarn.lock`,

> `pnpm-lock.yaml`, `bun.lock) without regenerating the lockfile from scratch:

Reviewer: ```suggestion
`pnpm-lock.yaml`, `bun.lock`) without regenerating the lockfile from scratch:
```

```bash

> chainctl libraries update-hashes

Reviewer 3: It's important to use this command in a directory containing lock file

Reviewer 3: or provide path to the file
